OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -125,14 +125,16 @@ public MultiplayerLogic(Widget widget, ModData modData, Action onStart, Action o
 
 			RefreshServerList();
 
-			if (directConnectHost != null)
+			if (directConnectAddress != null)
 			{
 				// The connection window must be opened at the end of the tick for the widget hierarchy to
 				// work out, but we also want to prevent the server browser from flashing visible for one tick.
 				widget.Visible = false;
 				Game.RunAfterTick(() =>
 				{
-					ConnectionLogic.Connect(directConnectHost, directConnectPort, "", OpenLobby, DoNothing);
+					if (directConnectAddress.IsValid)
+						ConnectionLogic.Connect(directConnectAddress, "", OpenLobby, DoNothing);
```

Comment:
```""``` is a bad idea you should use the ```String.Empty``` extension. not every platform or OS interprets "" as an empty string!